test/IRGen/distributed_actor_accessors.swift

```diff
@@ -0,0 +1,280 @@
+// RUN: %target-swift-frontend -emit-irgen %s -swift-version 5 -enable-experimental-distributed | %IRGenFileCheck %s
+
+// UNSUPPORTED: back_deploy_concurrency
+// REQUIRES: concurrency
+// REQUIRES: distributed
+
+import _Distributed
+
+enum SimpleE : Codable {
+case a
+}
+
+enum E : Codable {
+case a, b, c
+}
+
+enum IndirectE : Codable {
+  case empty
+  indirect case test(_: Int)
+}
+
+final class Obj : Codable, Sendable {
+  let x: Int
+
+  init(x: Int) {
+    self.x = x
+  }
+}
+
+struct LargeStruct : Codable {
+  var a: Int
+  var b: Int
+  var c: String
+  var d: Double
+}
+
+@available(SwiftStdlib 5.6, *)
+public distributed actor MyActor {
+  public typealias Transport = AnyActorTransport
+
+  distributed func simple1(_: Int) {
+  }
+
+  // `String` would be a direct result as a struct type
+  distributed func simple2(_: Int) -> String {
+    return ""
+  }
+
+  // `String` is an object that gets exploded into two parameters
+  distributed func simple3(_: String) -> Int {
+    return 42
+  }
+
+  // Enum with a single case are special because they have an empty
+  // native schema so they are dropped from parameters/result.
+  distributed func single_case_enum(_ e: SimpleE) -> SimpleE {
+    return e
+  }
+
+  distributed func with_indirect_enums(_: IndirectE, _: Int) -> IndirectE {
+    return .empty
+  }
+
+  // Combination of multiple arguments, reference type and indirect result
+  //
+  // Note: Tuple types cannot be used here is either position because they
+  // cannot conform to protocols.
+  distributed func complex(_: [Int], _: Obj, _: String?, _: LargeStruct) -> LargeStruct {
+    fatalError()
+  }
+}
```

Comment:
@DougGregor and @aschwaighofer Please take a look and let me know what else would be useful to add for the initial concrete-type only implementation. What ever it is has to be able to conform to Codable though...